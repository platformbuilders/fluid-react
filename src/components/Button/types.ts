import { ReactNode } from 'react';
import {
  ButtonColorType,
  ButtonVariants,
  TypographyVariants,
} from '@platformbuilders/theme-toolkit';
import RadixIcon from '@radix-ui/react-icons';

export type ButtonProps = {
  style?: any;
  type?: 'button' | 'submit' | 'reset';
  loading?: boolean;
  $fullWidth?: boolean;
  variant?: ButtonVariants;
  typographyVariant?: TypographyVariants;
  children?: string | ReactNode;
  leftIconName?: keyof typeof RadixIcon;
  rightIconName?: keyof typeof RadixIcon;
  colorVariant?: ButtonColorType;
  size?: 'normal' | 'medium' | 'large';
} & TouchableType;

export type TouchableType = {
  id?: string;
  accessibility: string;
  testID?: string;
  disabled?: boolean;
  onPress?: (param: any) => void;
};
