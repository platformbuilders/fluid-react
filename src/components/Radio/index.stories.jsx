import React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import markdown from './button.md';
import Button from './index';

storiesOf('Radio', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <Button variant="warning">{text('Text', 'Button Test')}</Button>
  ));
