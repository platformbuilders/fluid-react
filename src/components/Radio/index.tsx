import styled from 'styled-components';
import { FormControlLabel, Radio } from '@material-ui/core';
import { RadioProps } from '../../types';
import { getBackgroundColor, getHoverColor } from '../../utils';

export default styled(({ label, labelPlacement, value, ...rest }) => (
  <FormControlLabel
    control={<Radio {...rest} />}
    label={label}
    labelPlacement={labelPlacement}
    value={value}
  />
))<RadioProps>`
  &.MuiRadio-colorSecondary.Mui-checked {
    color: ${getBackgroundColor} !important;
    &:hover {
      background-color: ${getHoverColor} !important;
    }
  }

  &.MuiRadio-colorSecondary.Mui-checked + .MuiRadio-track {
    background-color: ${getBackgroundColor} !important;
  }
`;
