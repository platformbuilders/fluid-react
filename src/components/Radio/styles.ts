import styled from 'styled-components';

export const Wrapper = styled.div`
  margin: 16px 0;
`;

export const Input = styled.input`
  display: none;
  &:checked + label:before {
    border-color: rgb(51, 122, 183);
    animation: ripple 0.2s linear forwards;
  }
  &:checked + label:after {
    transform: scale(1);
  }
`;

export const Label = styled.label`
  display: inline-block;
  min-height: 20px;
  position: relative;
  padding: 0 30px;
  margin-bottom: 0;
  cursor: pointer;
  vertical-align: bottom;

  @keyframes ripple {
    0% {
      box-shadow: 0px 0px 0px 1px rgba(0, 0, 0, 0);
    }
    50% {
      box-shadow: 0px 0px 0px 15px rgba(0, 0, 0, 0.1);
    }
    100% {
      box-shadow: 0px 0px 0px 15px rgba(0, 0, 0, 0);
    }
  }

  &:before,
  &:after {
    position: absolute;
    content: '';
    border-radius: 50%;
    transition: all 0.3s ease;
    transition-property: transform, border-color;
  }

  &:before {
    left: 0;
    top: 0;
    width: 16px;
    height: 16px;
    border: 2px solid rgba(0, 0, 0, 0.54);
  }

  &:after {
    top: 5px;
    left: 5px;
    width: 10px;
    height: 10px;
    transform: scale(0);
    background: rgb(51, 122, 183);
  }
`;
