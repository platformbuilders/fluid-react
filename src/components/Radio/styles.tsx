import React from 'react';
import styled from 'styled-components';
import { FormControlLabel, Radio } from '@material-ui/core';

type Props = {
  id?: string;
  checked?: boolean;
  checkedIcon?: React.ReactNode;
  // variant?: ColorsVariant;
  disabled?: boolean;
  icon?: React.ReactNode;
  size?: 'small' | 'medium';
  label?: string;
  labelPlacement?: string;
};

export const StyledRadio = styled(({ label, labelPlacement, ...rest }) => (
  <FormControlLabel
    control={<Radio {...rest} />}
    label={label}
    labelPlacement={labelPlacement}
  />
))<Props>``;
