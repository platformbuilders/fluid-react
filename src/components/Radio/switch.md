---
id: component.radio
title: Radio
---

<!-- Component declaration begin -->

import { useState } from 'react';
import { Radio } from '@platformbuilders/react-elements';

export const RadioExample = () => {
const [isAnonymous, setIsAnonymous] = useState(false);

function handleIsAnonymous(e){
setIsAnonymous(e.target.value);
}

return(
<>
<Radio
label="Anônimo"
checked={isAnonymous}
onChange={handleIsAnonymous}
variant="secondary"
/>
<h3>{isAnonymous}</h3>
</>
);
}

<!-- Component declaration end -->

<!-- Documentation begin -->
Esse componente tem como função auxiliar em resposta com varias opção de resposta mas o uma sera valida

## Exemplo

### Fonte

```javascript
import { useState } from 'react';
import { Radio } from '@platformbuilders/react-elements';

export const RadioExample = () => {
const [isAnonymous, setIsAnonymous] = useState(false);

function handleIsAnonymous(e){
setIsAnonymous(e.target.checked);
}

return(
<>
<Radio
label="Anônimo"
value={isAnonymous}
onChange={handleIsAnonymous}
variant="secondary"
/>
<h3>{isAnonymous}</h3>
</>
);
}

```

## Atributos

| Formato        | Conceito      | Tipo   |
| ------|-----|-----|
| **value** 	| O valor é *any*. 	| **any** 	|
| **onChange**  	| é uma função que recebe por parametro event, para acessar o valor basta acessar *event.target.value* que retorna uma *string*. 	| **func**
| **label**  	| Aqui vai poder adicionar uma label ao lado do component 	| **boolean** 	|
| **labelPlacement**  	| Aqui pode definir a posição da label, por default é end	| **string = 'top' | 'start' | 'bottom' | 'end'** 	|
| **disabled**  	| Aqui vai desabilitar o componente 	| **boolean** 	|
| **variant** 	| Aqui sera usado as cores de acordo com theme, por padrão usa *primary*	| **string** 	|
| **checked** 	| ira marcar caso o retornor for true	| **boolean** 	|
| **size** 	| Ira determinar o tamanho do componente *small* ou *medium*	| **string** 	|
| **icon** 	| O icon mostrará quando o componente estiver unchecked	| **Node** 	|
| **checkedIcon** 	| O icon mostrará quando o componente estiver checked 	| **Node** 	|
