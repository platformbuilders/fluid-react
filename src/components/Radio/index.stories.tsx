import type { Meta, StoryObj } from '@storybook/react';
import Radio from './index';

const meta: Meta<typeof Radio> = {
  title: 'Radio',
  component: Radio,
  args: {
    label: 'An√¥nimo',
  },
};

export default meta;
type Story = StoryObj<typeof Radio>;

export const Default: Story = {};
