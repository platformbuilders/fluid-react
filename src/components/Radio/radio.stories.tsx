import { expect, jest } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { configure, userEvent, within } from '@storybook/testing-library';
import Radio from './index';

configure({ testIdAttribute: 'id' });

//  Mocks
const mockTextId = 'radio-test-id';
const mockOnChange = jest.fn();

const meta: Meta<typeof Radio> = {
  title: 'Components/Radio',
  component: Radio,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    id: mockTextId,
    label: 'Radio Test',
    group: 'radio-test',
    onChange: mockOnChange,
    disabled: false,
    checked: false,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Radio | Test Render', async () => {
      expect(canvas.getByTestId(mockTextId)).toBeInTheDocument();
    });

    await step('Radio | Event Click', async () => {
      await userEvent.click(canvas.getByTestId(mockTextId));
      expect(mockOnChange).toHaveBeenCalled();
    });
  },
};

type Story = StoryObj<typeof Radio>;

export const Default: Story = {
  args: {},
};

export default meta;
