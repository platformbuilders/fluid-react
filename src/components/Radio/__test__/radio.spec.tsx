import { shallow } from 'enzyme';
import faker from 'faker';
import { ThemeProvider } from 'styled-components';

import Radio from '..';
import { theme } from '../../../config/helpers';
import { RadioProps as Props } from '../../../types';

const defaultContent = 'Text';
const defaultProps: Props = {
  checked: true,
  label: defaultContent,
  onChange: jest.fn(),
};

describe('Component: Radio', () => {
  test('snapshots with default props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Radio {...defaultProps} />
      </ThemeProvider>,
    );
    expect(component).toMatchSnapshot();
  });

  test('snapshots with another props', () => {
    const valueMock = faker.random.words();
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Radio {...defaultProps} size="medium" value={valueMock} />
      </ThemeProvider>,
    );
    expect(component).toMatchSnapshot();
  });

  test('should call onChange when pressed', () => {
    // should
    const handleChange = jest.fn();

    // when
    const component = shallow(
      <Radio onChange={handleChange} variant="secondary" />,
    );

    // then
    expect(handleChange).not.toHaveBeenCalled();
    component.simulate('change');
    expect(handleChange).toHaveBeenCalled();
  });
});
