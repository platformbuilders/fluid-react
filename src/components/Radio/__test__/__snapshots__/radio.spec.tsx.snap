// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Radio snapshots with another props 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiRadio-root MuiRadio-colorSecondary sc-bczRLJ csSvjg PrivateSwitchBase-checked-9 Mui-checked MuiIconButton-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input-11"
          type="radio"
          value="TEXT"
        />
        <div
          class="PrivateRadioButtonIcon-root-12 PrivateRadioButtonIcon-checked-14"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </div>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Text
    </span>
  </label>
</div>
`;

exports[`Component: Radio snapshots with default props 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorSecondary sc-bczRLJ csSvjg PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorSecondary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input-4"
          type="radio"
          value=""
        />
        <div
          class="PrivateRadioButtonIcon-root-5 PrivateRadioButtonIcon-checked-7"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </div>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Text
    </span>
  </label>
</div>
`;
